src/prof.c

```diff
@@ -277,7 +277,8 @@ prof_sample_new_event_wait(tsd_t *tsd) {
 	 * otherwise bytes_until_sample would be 0 if u is exactly 1.0.
 	 */
 	uint64_t r = prng_lg_range_u64(tsd_prng_statep_get(tsd), 53);
-	double u = (r == 0U) ? 1.0 : (double)r * (1.0/9007199254740992.0L);
+	double u = (r == 0U) ? 1.0 : ((double)r *
+	    (1.0/9007199254740992.0));
```

Comment:
Doesn't look to be any different? Or am I missing something?

Reply:
I think C by default views a constant like `1.0` as double and `1.0L` as long double. The current version is all in double so it should be safe? Besides, the travis is failing but I cannot see any information regarding the error. Can you see anything from your side?

Reply:
I see. The division result is a tiny number so we may loss some precision if cast early / drop the long double. To be safe, maybe try casting after the multiply: `(double)(r * (1.0/9007199254740992.0L))` (might need convert r to long double?) I suspect this was the original intention.

Reply:
Sure, changed it back to the previous version where all are casted into long double and then cast the multiplication result back to double.